Scripts/extras/pull_and_build_from_git.py

```diff
@@ -593,30 +649,51 @@ def build_and_install_cmake(self):
                     shutil.copy2(glob_result, str(target_folder_path.resolve()), follow_symlinks=False)
 
+    def create_custom_env(self):
+        custom_env = os.environ.copy()
+        custom_env['TARGET_INSTALL_ROOT'] = str(self.build_install_folder.resolve())
+        custom_env['PACKAGE_ROOT'] = str(self.package_install_root.resolve())
+        custom_env['TEMP_FOLDER'] = str(self.base_temp_folder.resolve())
+        if self.package_info.depends_on_packages:
+            package_folder_list = ""
+            for package_name, _, subfoldername in self.package_info.depends_on_packages:
+                if package_folder_list:
+                    package_folder_list += ";"
+                # note that it must be an abs path for cmake to accept it:
+                # The package declaration may specify a subfolder name but the path library will
+                # behave correctly if you slash operator with an empty (by ignoring it)
+                package_folder_list += str( (self.base_temp_folder / package_name / subfoldername).resolve().absolute())
+            package_folder_list = package_folder_list.replace('\\', '/')
+            custom_env['DOWNLOADED_PACKAGE_FOLDERS'] = package_folder_list
+        return custom_env
+
     def build_and_install_custom(self):
         """
         Build and install from source using custom commands defined by 'custom_build_cmd' and 'custom_install_cmd'
         """
+        # we add TARGET_INSTALL_ROOT, TEMP_FOLDER and DOWNLOADED_PACKAGE_FOLDERS to the environ for both
+        # build and install, as they are useful to refer to from scripts.
+        
 
         custom_build_cmds = self.platform_config.get('custom_build_cmd', [])
         for custom_build_cmd in custom_build_cmds:
 
             call_result = subprocess.run(custom_build_cmd,
                                          shell=True,
                                          capture_output=False,
-                                         cwd=str(self.base_folder))
+                                         cwd=str(self.base_folder),
+                                         env=self.create_custom_env())
             if call_result.returncode != 0:
                 raise BuildError(f"Error executing custom build command {custom_build_cmd}")
 
         custom_install_cmds = self.platform_config.get('custom_install_cmd', [])
-        custom_install_env = os.environ.copy()
-        custom_install_env['TARGET_INSTALL_ROOT'] = str(self.build_install_folder.resolve())
+       
         for custom_install_cmd in custom_install_cmds:
             call_result = subprocess.run(custom_install_cmd,
                                          shell=True,
                                          capture_output=False,
                                          cwd=str(self.base_folder),
-                                         env=custom_install_env)
+                                         env=self.create_custom_env())
```

Comment:
Does `self.create_custom_env()` need to be repeated here, or are different 3P package configs allowed for `build` and `install`?

Reply:
no, its the same config.  I could remove the recreation of it.  The only danger is if something somehow modifies the env, so you get a clean env, but I'm not sure that's possible.